livekit-android-sdk/src/main/java/io/livekit/android/room/track/RemoteTrackPublication.kt

```diff
@@ -63,12 +64,15 @@ class RemoteTrackPublication(
     private var videoQuality: LivekitModels.VideoQuality? = LivekitModels.VideoQuality.HIGH
     private var videoDimensions: Track.Dimensions? = null
 
+    var subscriptionAllowed: Boolean = true
+        internal set
+
     val isAutoManaged: Boolean
         get() = (track as? RemoteVideoTrack)?.autoManageVideo ?: false
 
     override val subscribed: Boolean
         get() {
-            if (unsubscribed) {
+            if (unsubscribed || !subscriptionAllowed) {
```

Comment:
Wondering if this should return `false`. The track is subscribed to, but just that it does not have permissions. Feels like this needs to return an enum which gives a better picture of subscription state.

Reply:
hmm good question, I think technically it's desired to be subscribed, but tricky because clients use `subscribed` to indicate if subscription is successful and ready to be streamed. 

returning false probably makes more sense than true in this case.